### ISSUE TITLE
Make promotion mode more obvious in ui

### ISSUE THREAD

**Issue by kejne** [akuity/kargo#2778]
# Checklist

* [x] I've searched the issue queue to verify this is not a duplicate feature request.
* [x] I've pasted the output of `kargo version`, if applicable.
* [x] I've pasted logs, if applicable.

Kargo version tested: 0.9.1

# Proposed Feature

Make freights very easily selectable when promoting. When promoting freight, it should be obvious that you are in promotion mode and that selections at this point is to promote a freight. Other functionality should be hidden during this time to make it clear what is going on.

# Motivation

When promoting something currently, you really need to look hard that you're really in promotion-mode. Only a small "cancel" to the right and the text above says that you should select freight to promote. While it clearly has a new text, a nicer UX would be that the context is changed profoundly when the behavior of buttons also changes.

You will get a lot of tool tips when moving over the freight. The regular links are also highlighted, which confuses users that they will go somewhere else during the promotion and adds to the confusion if you are really promoting or not.

Moving cursor over freight when not promoting:
![image](https://github.com/user-attachments/assets/3e254230-8bec-4fb1-8240-e385e65e2df4)

Doing the same when promoting:
![image](https://github.com/user-attachments/assets/08602157-8e68-4492-822c-3aee052db7ac)
As can be seen these look basically identical.

# Suggested Implementation

During promotion, make everything around the freight line darker and highlight the freight line. Links within a freight (like copy commit sha, links to specific artifacts) should not be enabled. No tooltips and no underlining light-up. Instead the entire freight should be clickable to select the promotion only.

**Comment by krancour** [comment]
Agreed on most of that... although the tooltips might still be useful in identifying the Freight you're looking for. It would be unfortunate if you had to cancel the promotion workflow to inspect Freight more closely for the sake of identifying a particular one, only to then resume the promotion workflow. But I totally agree on improved visual cues and making the Freight less navigable/interactive during promotion. I have been frequently annoyed that when selecting Freight for promotion, one needs to be quite careful not to click on any UI element _within_ the Freight.

**Comment by Marvin9** [comment]
fixed in 1.5. we now add overlay in pipeline view in promotion mode
